roles/os_images/tasks/images.yml

```
@@ -83,8 +83,9 @@
 - name: Generate diskimage-builder images
   vars:
     dib_args: >-
-      {% if item.size is defined %}--image-size {{ item.size }}{% endif %} {% if item.type is defined %}-t {{ item.type }}{% endif %} {% if item.packages | default
-      %}-p {{ item.packages | join(',') }}{% endif %} {{ os_images_common }} {{ item.elements | join(' ') }} -o {{ item.name }}
+      {% if item.size is defined %}--image-size {{ item.size }}{% endif %} {% if item.type is defined %}-t {{ item.type }}{% endif %}
+      {% if item.packages | default %}-p {{ item.packages | join(',') }}{% endif %} {{ os_images_common }} {{ item.elements | join(' ') }}
+      -o {{ item.name }} -a {{ os_images_architecture }}
```

```suggestion
      -o {{ item.name }} {% if os_images_architecture is defined %}-a {{ os_images_architecture }}{% endif %}
```
I'd also be happy with this change, so it's left up to DIB if it's not explicitly set.

---

@m-bull's suggestion in the other comment thread is more correct.